app/crud/crud_pest_model.py

```diff
@@ -0,0 +1,57 @@
+from typing import Optional
+
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from crud.base import CRUDBase
+
+from models import PestModel, Cultivation
+from schemas import CreatePestModel, UpdatePestModel
+
+
+class CrudPestModel(CRUDBase[PestModel, CreatePestModel, UpdatePestModel]):
+
+    def create(self, db: Session, obj_in: CreatePestModel, **kwargs) -> Optional[PestModel]:
+
+        pest_model_db = PestModel(name=obj_in.name, description=obj_in.description, geo_areas_of_application=obj_in.geo_areas_of_application)
+
+        # Attempt to create a pest model entity
+        db.add(pest_model_db)
+        try:
+            db.commit()
+        except SQLAlchemyError:
+            db.rollback()
+            return None
+        db.refresh(pest_model_db)
+
+        # Assemble the cultivations from the list
+        cultivations = [Cultivation(name=x, pest_model_id=pest_model_db.id) for x in obj_in.cultivations]
+
+        # Attempt to create the cultivations and bind them to the previously created pest model entity
+        db.add_all(cultivations)
+        try:
+            db.commit()
```

Comment:
as in the Cultivation model there are not `unique` criteria for name and pest_model_id when is this going to fail?

Reply:
An example failure would be a connection issue with the DB, if it's deployed separately for example. 
It shouldn't fail in any 'normal' circumstances using the current docker deployment, but the commit can fail, even if it is pretty rare

Reply:
ok then